config/crd/bases/infrastructure.cluster.x-k8s.io_azuremanagedcontrolplanes.yaml

```diff
@@ -185,6 +230,51 @@ spec:
             description: AzureManagedControlPlaneSpec defines the desired state of
               AzureManagedControlPlane.
             properties:
+              aadProfile:
+                description: AadProfile is Azure Active Directory configuration to
+                  integrate with AKS for aad authentication.
+                properties:
+                  legacyAadProfile:
```

Comment:
I would almost be tempted to reject this for creates inside the reconciler? I would be okay allowing this for adoption scenarios, but we shouldn't be encouraging it for new clusters. 

That is a bit aggressive of a constraint, but I also explicitly didn't support Basic LB or VMAS. 

What do you think?

Reply:
It is a good point even I agree with that. I will make changes accordingly.